### ISSUE TITLE
Hyperparameter optimization with xgboost early_stopping_rounds support

### ISSUE THREAD

**Issue by gediminaszylius** [aws/sagemaker-python-sdk#654]
Hello,

Is there possibility to pass early_stopping_rounds when using hyperparameter optimization job with sagemaker?

I see there is generic early stopping for that (not specifically for xgboost) but it doesnt stop when model starts to overfit on training data even when validation data accuracy is dropping.

**Comment by yijiezz** [comment]
> Is there possibility to pass early_stopping_rounds when using hyperparameter optimization job with sagemaker?

Do you mean letting HPO job to tune parameter `early_stopping_rounds`? Currently SageMaker XGBoost does not support tuning `early_stopping_rounds`.
> I see there is generic early stopping for that (not specifically for xgboost) but it doesnt stop when model starts to overfit on training data even when validation data accuracy is dropping.

Do you mean a generic early stopping feature for HPO job? I don't think it is supported.

**Comment by yangaws** [comment]
Hi @gediminaszylius ,

I am going to close the issue. But if you have more questions, feel free to reopen. 

And if you want to make a feature request for what you ask, you can also reopen this issue with details of what you request and why it's important. 

Thanks!

**Comment by r-luo** [comment]
Hi @yijiezh, Does SageMaker XGBoost support `early_stopping_rounds during` fitting?
